### PR TITLE
fix(bulk): iterating through all attachments

### PR DESCRIPTION
**Pull request by jonahkaye**
Ticket: #[1793](https://github.com/metriport/metriport-internal/issues/1793)

### Description

- fixing a bug where we were not iterating through all attachments to find the documents to create presigned URLs for
- second issue where diverging implementations of search in core and api. a fair bit of complexity to jump through all the dependencies and make everything rely on core though, and don't want to potentially break other work flow and delay getting this to cx. Link to follow on [ticket](https://github.com/metriport/metriport-internal/issues/1801)

### Testing

- Staging
  - [x] branch to staging

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/aws/document-signing/bulk-sign.ts [hunk 1]
```diff
@@ -33,15 +33,11 @@ export async function getSignedUrls(
   try {
     const urls = await Promise.all(
       documents.map(async doc => {
-        // Check if content and attachment exist
-        if (!doc.content || !doc.content[0] || !doc.content[0].attachment) {
-          return;
-        }
-
-        const attachment = doc.content[0].attachment;
```
Comment (jonahkaye) [inline packages/core/src/external/aws/document-signing/bulk-sign.ts:41]: removing bug where only taking 0th index causing us to not get the attachment

File: packages/core/src/external/opensearch/search-documents.ts [hunk 2]
```diff
@@ -54,7 +57,8 @@ function checkExtensions(doc: DocumentReference) {
   if (!extensions) return false;
   const metriport = extensions.find(isMetriportExtension);
   const cw = extensions.find(isCommonwellExtension);
-  if (!metriport && !cw) return false;
+  const cq = extensions.find(isCarequalityExtension);
+  if (!metriport && !cw && !cq) return false;
```
Comment (jonahkaye) [inline packages/core/src/external/opensearch/search-documents.ts:61]: BUG! search-dcouments.ts code exists in core and in api, and it was only updated in api causing all CQ documents to not come up in search

File: packages/core/src/external/opensearch/search-documents.ts
Comment (jonahkaye) [inline packages/core/src/external/opensearch/search-documents.ts:1]: all changes here to make api and core look the same

File: packages/core/src/external/opensearch/index.ts [hunk 3]
```diff
@@ -11,3 +12,19 @@ export type IndexFields = {
   s3FileName: string;
   [contentFieldName]: string;
 };
+
+export function isDocStatusReady(doc: DocumentReference): boolean {
```
Comment (jonahkaye) [inline packages/core/src/external/opensearch/index.ts:16]: moved from api to core so it is now shared

**Review by Goncharo — APPROVED** [review]
:shipit:

**Pull request merged**